sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -295,7 +295,9 @@ class InMemoryCatalog(
     assert(tableDefinition.identifier.database.isDefined)
     val db = tableDefinition.identifier.database.get
     requireTableExists(db, tableDefinition.identifier.table)
-    catalog(db).tables(tableDefinition.identifier.table).table = tableDefinition
+    val updatedProperties = tableDefinition.properties.filter(kv => kv._1 != "comment")
+    val newTableDefinition = tableDefinition.copy(properties = updatedProperties)
+    catalog(db).tables(tableDefinition.identifier.table).table = newTableDefinition
```

Comment:
@gatorsmile @wzhfy  as i noticed for hive tables already the 'comment' property is been filtered while getting the hive table metadata(reading) in **HiveClientImpl.scala** -- > **getTableOption()** API. so i think it will be better if we can do the same for **InMemoryCatalogs** type table, while reading the table metadata we can exclude the 'comment', i think if we exclude from table definition while alter command then there can be an inconsistency, 
eg:  In create table we will persist the comment in metadata and while altering we are not.
Please correct me if i am going in wrong direction, Thanks.